Gemfile

```diff
@@ -1,8 +1,14 @@
 source 'https://rubygems.org'
 
-gem "github-pages", '193', group: :jekyll_plugins
+gem 'html-proofer'
```

Comment:
This isn't actually a build dependency, just a test one. Do we need to add it here? The typical user with ruby / jekyll will need to `bundle install` again. If most people don't need to run the tests locally, I would not add it.

Reply:
I ran the test locally and it reported 11 errors which are mostly related to newsletter

```shell

- ./_site/2020-02-10-newsletter/index.html
  *  External link http://carver.cs.ua.edu/RSE.htm failed: 404 No error
  *  External link https://cdh.princeton.edu/updates/2019/12/06/research-software-developer-job-opening/ failed: 404 No error
  *  External link https://main-princeton.icims.com/jobs/11125/performance-tuning-analyst/job failed: 410 No error
  *  External link https://www.psc.edu/about-psc/employment/3116-research-software-specialist-2014428 failed: 404 No error
- ./_site/2020-04-09-virtual-workshop/index.html
  *  internally linking to /events/2020/2020-04-virtual-workshop, which does not exist (line 450)
     <a href="/events/2020/2020-04-virtual-workshop">here</a>
- ./_site/2020-09-25-newsletter/index.html
  *  External link https://princeton.zoom.us/meeting/register/tJ0lce6hrjkqG9OyLUKYOfGDx1eb7OxzfYGK failed: 404 No error
  *  External link https://www.software.ac.uk/bloghttps://lbj20.blogspot.com/2020/09/software-maintenance-and-me.html failed: 404 No error
- ./_site/2020-11-30-newsletter/index.html
  *  External link https://iu.zoom.us/webinar/register/WN_MOKxyajxR4Cm0gEHHClIdg failed: 404 No error
  *  External link https://www.amazon.jobs/en/jobs/1276398/developer-advocate-for-hpc-batch failed: 404 No error
- ./_site/2021-02-16-newsletter/index.html
  *  External link https://jobs.lever.co/citrineinformatics/31a67fb7-105d-429a-8281-94ebf133825b failed: 404 No error
- ./_site/events/2020/2020-12-AGM.html
  *  External link https://princeton.zoom.us/meeting/register/tJErcuyrqj8rE9aUl9RwE4Y_-QetlSLK2BVj failed: 404 No error

HTML-Proofer found 11 failures!
```

And, these seem to be expired jobs. We need to update our Rakefile.
I do the run tests while giving my patch to any of the open-source projects.

And, I think we should encourage that too, else then our CI will have a backlog of failing PRs.

Reply:
For most of those links, they are in the newsletter folder, which we [ignore it](https://github.com/USRSE/usrse.github.io/blob/master/.github/workflows/urlchecker.yml#L30) with the urlchecker because a lot of links will no longer exist (old jobs or events). We also ignore zoom links, which won't exist. The virtual workshop link looks okay to me.